src/index.tsx

```diff
@@ -12,6 +12,7 @@ import { FiltersProvider } from './context/filtersContext/FiltersContext';
 import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
 import { ModalHookProvider } from './context/modalContext/ModalContext';
 import styled from 'styled-components';
+import ErrorPage from './404';
```

Comment:
In order to avoid the 404 page flashing before the homepage loads (See attached video), this component should be lazy loaded too. (Don't forget to change the component name as discussed). 

```suggestion
const ErrorPage = React.lazy(() =>
  import(/* webpackChunkName: "ErrorPage" */ './404')
);
```


https://user-images.githubusercontent.com/3723951/128843996-762e4cfd-a4ce-4adb-93e9-3429b978d100.mov